client/src/plus/ai-help/index.tsx

```diff
@@ -0,0 +1,411 @@
+            <form
+              ref={formRef}
+              className="ai-help-input-form"
+              onSubmit={(event) => {
+                event.preventDefault();
+                if (query.trim()) {
+                  submit(query.trim());
+                  setQuery("");
+                  setAutoScroll(true);
+                }
+              }}
+            >
+              <input
+                ref={inputRef}
+                disabled={isLoading || isResponding}
+                type="text"
+                onChange={(event) => setQuery(event.target.value)}
+                value={query}
+                placeholder={placeholder(
+                  isLoading
+                    ? "Requesting answer..."
+                    : isResponding
+                    ? "Receiving answer..."
+                    : "Ask your question"
+                )}
+              />
+              <Button
+                type="action"
+                icon="send"
+                buttonType="submit"
+                title="Submit question"
+                isDisabled={!query}
+              >
+                <span className="visually-hidden">Submit question</span>
+              </Button>
+            </form>
+            <div className="ai-help-footer-text">
+              <span>
+                Results based on MDN's most recent documentation and powered by
+                OpenAI GPT-3.5
+              </span>
+            </div>
+          </>
+        )}
+      </div>
+      {!hasConversation && !query && !isQuotaExceeded(quota) && (
+        <section className="ai-help-examples">
+          <header>Examples</header>
+          {EXAMPLES.map(({ category, query }, index) => (
+            <button
+              key={index}
+              type="button"
+              className={["ai-help-example", `category-${category}`].join(" ")}
+              onClick={() => {
+                setQuery(query);
+                inputRef.current?.focus();
+                window.setTimeout(() => window.scrollTo(0, 0));
+              }}
+            >
+              {query}
+            </button>
+          ))}
+        </section>
+      )}
+      <pre>{JSON.stringify(datas, null, 2)}</pre>
+    </section>
+  );
+}
+
+export function RoleIcon({ role }: { role: "user" | "assistant" }) {
+  const userData = useUserData();
+
+  switch (role) {
+    case "user":
+      return <Avatar userData={userData} />;
+
+    case "assistant":
+      return <Icon name="chatgpt" />;
+  }
+}
+
+function useAutoScroll(dependency) {
+  const [autoScroll, setAutoScroll] = useState(true);
+  const bodyRef = useRef<HTMLElement | null>();
+  const footerRef = useRef<HTMLElement | null>();
+  const lastScrollY = useRef(0);
+  const lastHeight = useRef(0);
+
+  useEffect(() => {
+    const body = (bodyRef.current ??=
+      document.querySelector<HTMLElement>(".ai-help-body"));
+    const footer = (footerRef.current ??=
+      document.querySelector<HTMLElement>(".ai-help-footer"));
```

Comment:
Not sure if it's the cause of the autoscroll bug (which I've been unable to reproduce yet), but generally it's best to set DOM refs with `ref={foobar}` on the element, something like this (and following suggestions):

```suggestion
function useAutoScroll(bodyRef: React.RefObject<HTMLElement>, footerRef: React.RefObject<HTMLElement>, dependency: any) {
  const [autoScroll, setAutoScroll] = useState(true);
  const lastScrollY = useRef(0);
  const lastHeight = useRef(0);

  useEffect(() => {
    const body = bodyRef.current;
    const footer = footerRef.current;
```